### PR TITLE
Fix to avoid REST API calls at log level 2.

### PR DESCRIPTION
**Pull request by aveshagarwal**
By default, logging REST API calls at log level 2 is too verbose (see below), if log level 2 happens to be default. So increasing its default to 3.

```
I0203 12:37:27.059555   11483 wrap.go:75] PUT /api/v1/namespaces/kube-system/endpoints/kube-scheduler: (1.358954ms) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/leader-election] [::1]:35168]
I0203 12:37:27.780029   11483 wrap.go:75] GET /api/v1/nodes/127.0.0.1?resourceVersion=0: (819.444µs) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format] [::1]:35154]
I0203 12:37:27.844846   11483 wrap.go:75] POST /api/v1/namespaces/default/events: (11.337447ms) 201 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format] [::1]:35154]
I0203 12:37:27.844851   11483 wrap.go:75] PATCH /api/v1/nodes/127.0.0.1/status: (9.998981ms) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format] [::1]:35154]
I0203 12:37:28.942863   11483 wrap.go:75] GET /api/v1/namespaces/kube-system/endpoints/kube-controller-manager: (923.711µs) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/leader-election] [::1]:35156]
I0203 12:37:28.944556   11483 wrap.go:75] PUT /api/v1/namespaces/kube-system/endpoints/kube-controller-manager: (1.188942ms) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/leader-election] [::1]:35156]
I0203 12:37:29.061483   11483 wrap.go:75] GET /api/v1/namespaces/kube-system/endpoints/kube-scheduler: (937.549µs) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/leader-election] [::1]:35168]
I0203 12:37:29.063068   11483 wrap.go:75] PUT /api/v1/namespaces/kube-system/endpoints/kube-scheduler: (1.111312ms) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/leader-election] [::1]:35168]
I0203 12:37:30.947922   11483 wrap.go:75] GET /api/v1/namespaces/kube-system/endpoints/kube-controller-manager: (935.198µs) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/leader-election] [::1]:35156]
I0203 12:37:30.950150   11483 wrap.go:75] PUT /api/v1/namespaces/kube-system/endpoints/kube-controller-manager: (1.703438ms) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/leader-election] [::1]:35156]
I0203 12:37:31.064883   11483 wrap.go:75] GET /api/v1/namespaces/kube-system/endpoints/kube-scheduler: (1.127992ms) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/leader-election] [::1]:35168]
I0203 12:37:31.066503   11483 wrap.go:75] PUT /api/v1/namespaces/kube-system/endpoints/kube-scheduler: (1.098029ms) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/leader-election] [::1]:35168]
I0203 12:37:32.951691   11483 wrap.go:75] GET /api/v1/namespaces/kube-system/endpoints/kube-controller-manager: (945.295µs) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/leader-election] [::1]:35156]
I0203 12:37:32.953580   11483 wrap.go:75] PUT /api/v1/namespaces/kube-system/endpoints/kube-controller-manager: (1.331822ms) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/leader-election] [::1]:35156]
I0203 12:37:33.068221   11483 wrap.go:75] GET /api/v1/namespaces/kube-system/endpoints/kube-scheduler: (912.121µs) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/leader-election] [::1]:35168]
I0203 12:37:33.069787   11483 wrap.go:75] PUT /api/v1/namespaces/kube-system/endpoints/kube-scheduler: (1.120666ms) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/leader-election] [::1]:35168]
I0203 12:37:34.955546   11483 wrap.go:75] GET /api/v1/namespaces/kube-system/endpoints/kube-controller-manager: (1.02279ms) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/leader-election] [::1]:35156]
I0203 12:37:34.957812   11483 wrap.go:75] PUT /api/v1/namespaces/kube-system/endpoints/kube-controller-manager: (1.661017ms) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/leader-election] [::1]:35156]
I0203 12:37:35.071528   11483 wrap.go:75] GET /api/v1/namespaces/kube-system/endpoints/kube-scheduler: (935.155µs) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/leader-election] [::1]:35168]
I0203 12:37:35.073087   11483 wrap.go:75] PUT /api/v1/namespaces/kube-system/endpoints/kube-scheduler: (1.027371ms) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/leader-election] [::1]:35168]
I0203 12:37:35.580075   11483 wrap.go:75] GET /apis/extensions/v1beta1/thirdpartyresources: (943.453µs) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format] [::1]:35144]
I0203 12:37:36.611659   11483 wrap.go:75] GET /api/v1/namespaces/default: (753.781µs) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format] [::1]:35144]
I0203 12:37:36.612516   11483 wrap.go:75] GET /api/v1/namespaces/default/services/kubernetes: (495.105µs) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format] [::1]:35144]
I0203 12:37:36.613167   11483 wrap.go:75] GET /api/v1/namespaces/default/endpoints/kubernetes: (379.568µs) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format] [::1]:35144]
I0203 12:37:36.960131   11483 wrap.go:75] GET /api/v1/namespaces/kube-system/endpoints/kube-controller-manager: (1.431137ms) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/leader-election] [::1]:35156]
I0203 12:37:36.963470   11483 wrap.go:75] PUT /api/v1/namespaces/kube-system/endpoints/kube-controller-manager: (2.190438ms) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/leader-election] [::1]:35156]
I0203 12:37:37.028185   11483 wrap.go:75] GET /api/v1/nodes: (1.34149ms) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/pod-garbage-collector] [::1]:35156]
I0203 12:37:37.074666   11483 wrap.go:75] GET /api/v1/namespaces/kube-system/endpoints/kube-scheduler: (928.261µs) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/leader-election] [::1]:35168]
I0203 12:37:37.076314   11483 wrap.go:75] PUT /api/v1/namespaces/kube-system/endpoints/kube-scheduler: (1.240852ms) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/leader-election] [::1]:35168]
I0203 12:37:37.847163   11483 wrap.go:75] GET /api/v1/nodes/127.0.0.1?resourceVersion=0: (725.021µs) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format] [::1]:35154]
I0203 12:37:37.901326   11483 wrap.go:75] PATCH /api/v1/nodes/127.0.0.1/status: (2.377445ms) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format] [::1]:35154]
I0203 12:37:38.968028   11483 wrap.go:75] GET /api/v1/namespaces/kube-system/endpoints/kube-controller-manager: (3.777083ms) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/leader-election] [::1]:35156]
I0203 12:37:38.970313   11483 wrap.go:75] PUT /api/v1/namespaces/kube-system/endpoints/kube-controller-manager: (1.655815ms) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/leader-election] [::1]:35156]
I0203 12:37:39.077913   11483 wrap.go:75] GET /api/v1/namespaces/kube-system/endpoints/kube-scheduler: (919.65µs) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/leader-election] [::1]:35168]
I0203 12:37:39.079617   11483 wrap.go:75] PUT /api/v1/namespaces/kube-system/endpoints/kube-scheduler: (1.196855ms) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/leader-election] [::1]:35168]
I0203 12:37:40.972171   11483 wrap.go:75] GET /api/v1/namespaces/kube-system/endpoints/kube-controller-manager: (936.07µs) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/leader-election] [::1]:35156]
I0203 12:37:40.973886   11483 wrap.go:75] PUT /api/v1/namespaces/kube-system/endpoints/kube-controller-manager: (1.200048ms) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/leader-election] [::1]:35156]
I0203 12:37:41.084158   11483 wrap.go:75] GET /api/v1/namespaces/kube-system/endpoints/kube-scheduler: (3.842758ms) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/leader-election] [::1]:35168]
I0203 12:37:41.085722   11483 wrap.go:75] PUT /api/v1/namespaces/kube-system/endpoints/kube-scheduler: (1.101371ms) 200 [[hyperkube/v0.0.0 (linux/amd64) kubernetes/$Format/leader-election] [
```

xref: https://bugzilla.redhat.com/show_bug.cgi?id=1414813

@kubernetes/rh-cluster-infra 

Fix https://github.com/kubernetes/kubernetes/issues/47916

### CODE REVIEW

**Comment by eparis** [comment]
was this a change since 1.4?

**Comment by eparis** [comment]
@smarterclayton looks like you originally picked v(2) way back in 4e56dafecc9d609f45bb4577a2c30c173d17964e

https://github.com/kubernetes/kubernetes/pull/1359

Who would need to approve moving to V(3)

**Comment by smarterclayton** [comment]
@lavalamp because we talked about this being an operational tool for normal function, and I suspect people rely on it in GKE and others at v2 today to get logging.  Changing to v3 would break anyone who scrapes at v2.

That said, I agree this is very verbose.

**Comment by eparis** [comment]
on a cluster of any size doing work it seems to be swampingly verbose  :)  whereas the other v(2) stuff isn't so bad...

**Comment by smarterclayton** [comment]
My gut would be that it should be easy to opt out of this and that most people most of the time don't need this (which is what 3 was for).  But because it has operational impact...

**Comment by gmarek** [comment]
cc @wojtek-t 

I'm leaning against this PR. Those are probably most useful logs in the whole system - I look at them pretty much every time I'm debugging anything. I agree that they are verbose, but they're not for direct human consumption, not enhanced by the power for grep.

@aveshagarwal - why do you want to make this change? Is the size of logs problem?

**Comment by aveshagarwal** [comment]
@gmarek I agree these are useful logs, but the question is do we want these logs at --v=2?

**Comment by gmarek** [comment]
I think that we do. Logs are there to help figure out what's wrong, and those logs are extremely important for that. If someone want no-logging (aka 'I don't care') there's always V=0.

**Comment by smarterclayton** [comment]
v(2) is defined as "every user wants them".  If we can make a case that
every kube admin wants these logs by default, v=2 is appropriate.

**Comment by gmarek** [comment]
I do think that 'every (sane) user wants them', but that's not really a formal definition:)

**Comment by gmarek** [comment]
In other words - I think we should close this PR.

**Comment by mffiedler** [comment]
Level 2 usually provides enough info for devs to start debug process.   Having that level sane really helps when difficult  to repro problems are encountered.

**Comment by smarterclayton** [comment]
We've caught a fair number of panics, crashes, bad behaviors (eaten
errors), and otherwise in test suites by simple observation of the log.
Agree it's not quite the level of volume I would expect at V(2), but it's
still possible to tune it down with --vmodule.

**Comment by gmarek** [comment]
I'm closing this PR. Ping me if you disagree.

**Comment by ncdc** [comment]
@gmarek we have been discussing this again today. I agree with you that these logs are useful when debugging issues. However, they have a negative performance impact when running large clusters. Would you think it's reasonable to proceed with this, with the expectation that for production clusters, admins would increase the loglevel from 2 to 3 or 4 if they need to debug issues? Otherwise, 2 would be an expected default for informational log messages but not spammy ones. Also, for PR jobs, we should be running at 3 or 4 if we're not already. WDYT?

**Comment by gmarek** [comment]
No, level 3 is already too verbose. I understand the performance implications, but maybe we could get past them by dealing with `--logtostderr` flag, which is most likely main culprit (unbuffered writes)? I don't know what/if golang team is doing with glog package, so I don't know if it's possible...

Also, this is just my (admittedly very strong) opinion, so if enough important people agree that we should do that, then I won't be able to block this (though honestly - I'm pretty convinced that it'll cost you more debugging time, that you'll save on CPU).

Can't you run cluster with `--v=1`?

@wojtek-t @smarterclayton @thockin @lavalamp

**Comment by ncdc** [comment]
@gmarek if we run the cluster at `--v=1` and run into an issue, we almost always have to bump up to `--v=2` (or higher) to truly debug. I suppose another option is to move the non-spammy messages we care about that are currently at 2 down to 1. But from what I've heard from @smarterclayton, we currently want to run at 2 by default. We are trying to eliminate overly spammy messages from 2.

My suggestion is that we move this to 3, run clusters at 2, and if there is an issue that needs debugging, you restart the components at 3 and try to reproduce the issue, if that's acceptable.

**Comment by gmarek** [comment]
The thing is that you never know when issue will surface, and when to do this.

**Comment by eparis** [comment]
For the most part I feel like v(2) has informational changes about a cluster that an admin cares about. v(3) has engineering debug stuff. This is CLEARLY engineering debug stuff. Practically no admin wants to be spammed with these. Real life example on a small cluster (7 nodes with 190 pods) we see over 82,000 messages per hour from this. And less than 4000 messages per hour for not this.

This one spam accounts for x20 more messages than the rest of V(2) put together.

**Comment by ncdc** [comment]
@gmarek it's a balance, and it's one that's hard to win :-) You either spam the cluster with logs that you don't need 99% of the time, or you don't and it's harder to debug the past.

**Comment by wojtek-t** [comment]
In my personal opinion, those logs are in fact kind of "audit logs" (because those are exactly about API calls). That said, in the ideal world those should just go to a different location/file.

I generally agree that cluster admins may treat them more like a spam. But at the same time, those are super extremely useful for engineers. So as long as I'm fine with moving them out of V(2), I'm really against about putting them into V(3) (where there are (or at least were in the past) thousands of logs that I've never used and thought that I will use for anything).

That said, maybe as some kind of consensus is to make the flag-gated behind some other flag? Or something like that?
In other words, I think it's reasonable request to disable those by default, but I would like to be able to switch them on without switching on anything else.
@ncdc @eparis @gmarek - WDYT?

**Comment by ncdc** [comment]
@wojtek-t I'm not opposed to being able to enable/disable certain log messages. I would hate to add a flag/gate just for this without thinking through a reasonable design, as we'll probably have a similar situation with some other log messages in the future. Maybe we could find something other than `glog` that has this functionality?

**Comment by gmarek** [comment]
Second level flag gating (as verbosity level is already a flag gate) doesn't sound particularly compelling, but it'll do the trick. I'm more inclined to clean up `V(3)`, and/or prioritizing making buffered writes work to get rid of the root cause once and for all, but if everything we can afford is another flag, then let it be.

**Comment by eparis** [comment]
Would it be possible to move it to V(3) and then, if people want it, they can just `--v=2 --vmodule='panics=v3'` ? Which I *think* would have v2 + just this message?

**Comment by wojtek-t** [comment]
> @wojtek-t I'm not opposed to being able to enable/disable certain log messages. I would hate to add a flag/gate just for this without thinking through a reasonable design, as we'll probably have a similar situation with some other log messages in the future. Maybe we could find something other than glog that has this functionality?

@ncdc I agree a dedicated flag is by far not perfect solution. If there is some other solution for it, it would be great.
Though, i don't fully agree that we will have a lot of similar situations like this in the future. These logs are pretty special - we are logging API calls in them. So if we have a dedicated mechanism for audit logs ( what's the status of - I think it was planned, but I didn't follow it at all, @smarterclayton - I guess you will know this), then those logs won't be useful at all.

**Comment by wojtek-t** [comment]
> Would it be possible to move it to V(3) and then, if people want it, they can just --v=2 --vmodule='panics=v3' ? Which I think would have v2 + just this message?

@eparis - I think I don't understand. Can you clarify?

**Comment by soltysh** [comment]
> So if we have a dedicated mechanism for audit logs ( what's the status of - I think it was planned, but I didn't follow it at all, @smarterclayton - I guess you will know this), then those logs won't be useful at all.

@wojtek-t here's the current state of audit and what you can get out of it. More is planned and hopefully will be discussed during today's sig-api-machinery call

**Comment by soltysh** [comment]
The links is https://kubernetes.io/docs/concepts/cluster-administration/audit/

**Comment by ncdc** [comment]
Given that this is essentially a webserver access log, I think a flag to enable/disable and maybe one to output to a separate log file (if that's desirable) would be the way to go. @wojtek-t I agree with you and I'll retract my suggestion of needing this in the future for a lot of other log messages.

**Comment by eparis** [comment]
@wojtek-t I'm going to update my old comment to fix a mistake and explain it here.

The log message showed it as "wrap.go:75". So I was suggesting we turn on V(3) for "wrap".
2 problems with that comment.
1. I think the message is actually coming from log.go, not wrap.go.
2. I don't know if vmodule just uses the filename or the path/filename. If it is just the filename having to turn on v(3) for "log" could pick up other messages we didn't want.

I guess I can play with glog a bit and come back with intelligent thoughts rather than wild conjecture.

**Comment by wojtek-t** [comment]
@soltysh - thanks for the link; that's pretty much exactly what we need. Once we have this, we should get rid of the logs we are talking about in favor of audit logging.

That said, we should solve this problem with as low cost as possible. I think the suggestion made by @eparis is pretty good - if we can tune it to do exactly what we want, I'm fine with proceeding with it.

**Comment by eparis** [comment]
I wrote https://github.com/eparis/glog-test.

I think you would have to use `--v=2 --vmodule="log=3" to get these. No idea what other V(3)'s live in a file called 'log.go'

**Comment by ncdc** [comment]
So if we can figure out how to get the file printed by glog to be something unique (accesslog.go or similar), maybe this would work?

**Comment by eparis** [comment]
Actually, it doesn't matter at the moment.
```
for file in $(find . -name log.go); do
  echo "looking at ${file}";
  grep 'glog.V(3)' "${file}";
done
```
Finds no v(3) users in and log.go file

**Comment by eparis** [comment]
Since we know how to selectively enable/disable this are we all agreed this PR is G2G? Someone will have to make the `vmodule` change on GKE for 1.7, no idea where that lives.

**Comment by eparis** [comment]
@wojtek-t @gmarek is there something that can be done in this PR to make GKE configured the way you want it? Or is that something that is done behind closed doors?

**Comment by gmarek** [comment]
cc @Kashomon @jlowdermilk

**Comment by wojtek-t** [comment]
> @wojtek-t @gmarek is there something that can be done in this PR to make GKE configured the way you want it? Or is that something that is done behind closed doors?

Sorry I missed this update.
Can we change our tests to output it by default before doing this change?

**Comment by marun** [comment]
@aveshagarwal Here are some potential places to set the log level:

apiserver log level: https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/config-test.sh#L129
apiserver test args: https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/config-test.sh#L142

cc: @eparis

**Comment by aveshagarwal** [comment]
@marun let me look.

**Comment by aveshagarwal** [comment]
@marun  It seems that cluster/gce/config-test.sh already has log level --v=4.  So I dont think so it would make any difference.

It seems that ./cluster/kubemark/gce/config-default.sh has  log  level at --v=2, wonder if I need to change this file?

**Comment by eparis** [comment]
@aveshagarwal I'd say that's up to @wojtek-t

**Comment by aveshagarwal** [comment]
@eparis ok not going to change that. Throwing other options:

There are several files using log level 2, but some that seem relevant ones are:

./cluster/gce/container-linux/configure-helper.sh:
./cluster/gce/gci/configure-helper.sh
./cluster/saltbase/pillar/logging.sls
./cluster/get-kube-local.sh

Need to change any of these?

**Comment by marun** [comment]
@aveshagarwal So, maybe nothing needs to be done for the tests, the log level is already high enough?

**Comment by aveshagarwal** [comment]
please keep it open.

**Comment by brendandburns** [comment]
/lgtm

/release-note-none

LGTM + release-note-none from the bulk LGTM tool

**Comment by aveshagarwal** [comment]
Create issue https://github.com/kubernetes/kubernetes/issues/47916 for it.

**Comment by lavalamp** [comment]
@brendandburns I don't think debate had settled on this PR, this was probably not a good candidate for a bulk LGTM

**Comment by lavalamp** [comment]
(Sorry, I didn't see this PR until just now)

IMO seeing rest calls in apiserver's log is how you know the system is working if you are ssh'd in trying to fix something. If we turn those off, there needs to be a summary that does stay on (e.g. "XX REST calls in the last YY time period").

**Comment by lavalamp** [comment]
Also, glog is very flexible: https://github.com/golang/glog/blob/master/glog.go#L55-L69

It should be easy for users to not see these messages even if they leave it at v=2 with the vmodule flag.

**Comment by brendandburns** [comment]
Yeah, I should probably add some logic around "if there hasn't been any discussion in > 2 days" or somesuch to the bulk lgtm...


Sorry for the noise.


--brendan



________________________________

**Comment by eparis** [comment]
@lavalamp `IMO seeing rest calls in apiserver's log is how you know the system is working if you are ssh'd in trying to fix something.`

This is what metrics are for.  If we are not exposing in metrics what is happening we should fix that. V(2) should not be hundreds of messages per second saying 'nothing to see here'. If the message says anything about a problem, and action an operator should pay attention to, a change in the system we want written down for posterity, whatever, it makes sense in V(2).

But right now this is gigs of data you have to sort out to actually see interesting things in the system.

glog, while flexible, requires a restart to make changes. If it didn't require a restart I'd like it a lot more. Would Google take changes to glog? Even so, I would argue that messages which don't say something meaningful don't belong in V(2).

**Comment by brendandburns** [comment]
@lavalamp Do we have consensus on this? If we think it is spammy to most people it probably shouldn't be `v(2)`

**Comment by gmarek** [comment]
@brendandburns - I think it needs some work before it can be done and a wider discussion. Number of people depend on those logs when debugging, so silently removing them is not a way to go. Additionally we probably need to bump all verbosity levels in API server by one, so that it'll be possible to keep current behavior, without enormous amount of spam that (possibly) V(3) would involve. There's not a huge amount of V(3) logs, so audit should be possible to do in few hours. @kubernetes/sig-api-machinery-feature-requests

**Comment by eparis** [comment]
@gmarek What problem(s) was debugged with these logs? I'd love any examples people can remember why they needed these messages and couldn't up the log level/restart to get them.

I also *smile snidely smile* point out you are saying these are debugging messages at a non-debug level of logging   :)

On a medium sized cluster we are talking literally gigs per day of useless debug spam that appears to serve no purpose other than what @lavalamp stated "a way to see the api server is working when ssh'd in". While I agree with Daniel, it does that, I do not find it a convincing argument why it should stay for everyone.

**Comment by gmarek** [comment]
Number of cases - whenever I need to check if some requests are received by the API server. Or to check what deleted/created/modified given object.

**Comment by eparis** [comment]
On Mon, 2017-09-18 at 16:40 +0000, Marek Grabowski wrote:
> Number of cases - whenever I need to check if some requests are
> received by the API server. Or to check what deleted/created/modified
> given object.

Sounds like you are describing the audit log. Not what belongs in the
API server log:

https://kubernetes.io/docs/tasks/debug-application-cluster/audit/

No?

**Comment by crassirostris** [comment]
BTW, audit logs are currently written to a log file by default

**Comment by gmarek** [comment]
Audit log is fine I guess. But IIUC it'll contain exactly the same information you're complaining about, so the problem will not go away, it'll just change the name.

**Comment by brendandburns** [comment]
@gmarek @eparis do we have a conclusion here?

**Comment by cblecker** [comment]
This appears to be somewhat active. Removing the lifecycle/frozen state.

/remove-lifecycle frozen

**Comment by liggitt** [comment]
> But IIUC it'll contain exactly the same information you're complaining about, so the problem will not go away, it'll just change the name.

audit lets you select some resources and not others, and allows for async/batch processing to avoid bottlenecks

I'm in favor of this change

**Comment by lavalamp** [comment]
....I was writing a comment, but the bot merged this, apparently on a phantom approval.

I think changing this deserved publication on sigs for people who run clusters.

**Comment by wojtek-t** [comment]
@shyamjvs @kubernetes/sig-scalability-bugs 

I don't think we should have merged that - this will significantly affect our debugging ability (at least in the context of performance tests, where we can't really run tests on higher verbosity, since the amount of logs there is overloading apiserver).

So I think we may consider reverting it, and making verbosity for those logs a flag (I'm fine with having a default still at 3, but I would like to be able to modify that for the purpose of scalability tests).

**Comment by ncdc** [comment]
@shyamjvs please see the `-vmodule` comments in #62468. We probably don't need to revert this if it's acceptable to adjust the scalability logging flags.

**Comment by eparis** [comment]
I think Andy is right and this PR is right, -vmodule can make these logs available but cannot be used to hide them. vmodule only goes in 1 direction.

**Comment by shyamjvs** [comment]
I've sent a PR enabling the logs through -vmodule for our tests.
However, there's still a discussion whether it should be a prod default under https://github.com/kubernetes/kubernetes/issues/62468#issuecomment-380898675.
